dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/imports/ParamsConverterTest.java

```diff
@@ -0,0 +1,268 @@
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.tracker.imports;
+
+import static org.hisp.dhis.test.utils.Assertions.assertContainsOnly;
+import static org.hisp.dhis.test.utils.Assertions.assertIsEmpty;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+
+import java.util.List;
+import org.hisp.dhis.common.UID;
+import org.hisp.dhis.program.Event;
+import org.hisp.dhis.program.Program;
+import org.hisp.dhis.program.ProgramStage;
+import org.hisp.dhis.program.ProgramType;
+import org.hisp.dhis.test.TestBase;
+import org.hisp.dhis.tracker.imports.bundle.TrackerBundle;
+import org.hisp.dhis.tracker.imports.domain.Enrollment;
+import org.hisp.dhis.tracker.imports.domain.MetadataIdentifier;
+import org.hisp.dhis.tracker.imports.domain.Relationship;
+import org.hisp.dhis.tracker.imports.domain.TrackedEntity;
+import org.hisp.dhis.tracker.imports.domain.TrackerDto;
+import org.hisp.dhis.tracker.imports.domain.TrackerEvent;
+import org.hisp.dhis.tracker.imports.domain.TrackerObjects;
+import org.hisp.dhis.tracker.imports.preheat.TrackerPreheat;
+import org.hisp.dhis.user.User;
+import org.hisp.dhis.user.UserDetails;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class ParamsConverterTest extends TestBase {
+  private static final UID TRACKER_EVENT_UID = UID.generate();
+
+  private static final UID SINGLE_EVENT_UID = UID.generate();
+
+  private Program programWithRegistration;
+
+  private Program programWithoutRegistration;
+
+  private ProgramStage programStageWithRegistration;
+
+  private ProgramStage programStageWithoutRegistration;
+
+  private UserDetails user;
+
+  @Mock private TrackerPreheat trackerPreheat;
+
+  @BeforeEach
+  void setup() {
+    user = UserDetails.fromUser(new User());
+
+    programWithRegistration = createProgram('A');
+    programWithRegistration.setProgramType(ProgramType.WITH_REGISTRATION);
+    programWithoutRegistration = createProgram('B');
+    programWithoutRegistration.setProgramType(ProgramType.WITHOUT_REGISTRATION);
+    programStageWithRegistration = createProgramStage('C', programWithRegistration);
+    programStageWithoutRegistration = createProgramStage('D', programWithoutRegistration);
+  }
+
+  @Test
+  void shouldSuccessToConvert() {
```

Comment:
Would you add the 'when' part in the test name?
I'm not really sure where do the events com from here, are they empty?
What about relationships? Are they empty too?